editor/plugins/theme_editor_plugin.cpp

```diff
@@ -34,12 +34,1186 @@
+
+					item_list = &tree_color_items;
+					color_amount += filtered_names.size();
+					break;
+
+				case Theme::DATA_TYPE_CONSTANT:
+					data_type_node->set_icon(0, get_theme_icon("MemberConstant", "EditorIcons"));
+					data_type_node->set_text(0, TTR("Constants"));
+
+					item_list = &tree_constant_items;
+					constant_amount += filtered_names.size();
+					break;
+
+				case Theme::DATA_TYPE_FONT:
+					data_type_node->set_icon(0, get_theme_icon("Font", "EditorIcons"));
+					data_type_node->set_text(0, TTR("Fonts"));
+
+					item_list = &tree_font_items;
+					font_amount += filtered_names.size();
+					break;
+
+				case Theme::DATA_TYPE_FONT_SIZE:
+					data_type_node->set_icon(0, get_theme_icon("FontSize", "EditorIcons"));
+					data_type_node->set_text(0, TTR("Font Sizes"));
+
+					item_list = &tree_font_size_items;
+					font_size_amount += filtered_names.size();
+					break;
+
+				case Theme::DATA_TYPE_ICON:
+					data_type_node->set_icon(0, get_theme_icon("ImageTexture", "EditorIcons"));
+					data_type_node->set_text(0, TTR("Icons"));
+
+					item_list = &tree_icon_items;
+					icon_amount += filtered_names.size();
+					break;
+
+				case Theme::DATA_TYPE_STYLEBOX:
+					data_type_node->set_icon(0, get_theme_icon("StyleBoxFlat", "EditorIcons"));
+					data_type_node->set_text(0, TTR("Styleboxes"));
+
+					item_list = &tree_stylebox_items;
+					stylebox_amount += filtered_names.size();
+					break;
+
+				case Theme::DATA_TYPE_MAX:
+					break; // Can't happen, but silences warning.
+			}
+
+			bool data_type_any_checked = false;
+			bool data_type_any_checked_with_data = false;
+
+			filtered_names.sort_custom<StringName::AlphCompare>();
+			for (List<StringName>::Element *F = filtered_names.front(); F; F = F->next()) {
+				TreeItem *item_node = import_items_tree->create_item(data_type_node);
+				item_node->set_meta("_can_be_imported", true);
+				item_node->set_text(0, F->get());
+				item_node->set_cell_mode(IMPORT_ITEM, TreeItem::CELL_MODE_CHECK);
+				item_node->set_checked(IMPORT_ITEM, false);
+				item_node->set_editable(IMPORT_ITEM, true);
+				item_node->set_cell_mode(IMPORT_ITEM_DATA, TreeItem::CELL_MODE_CHECK);
+				item_node->set_checked(IMPORT_ITEM_DATA, false);
+				item_node->set_editable(IMPORT_ITEM_DATA, true);
+
+				_restore_selected_item(item_node);
+				if (item_node->is_checked(IMPORT_ITEM)) {
+					data_type_any_checked = true;
+					any_checked = true;
+				}
+				if (item_node->is_checked(IMPORT_ITEM_DATA)) {
+					data_type_any_checked_with_data = true;
+					any_checked_with_data = true;
+				}
+
+				item_list->push_back(item_node);
+			}
+
+			data_type_node->set_checked(IMPORT_ITEM, data_type_any_checked);
+			data_type_node->set_checked(IMPORT_ITEM_DATA, data_type_any_checked && data_type_any_checked_with_data);
+		}
+
+		// Remove the item if it doesn't match the filter in any way.
+		if (!is_matching_filter && !has_filtered_items) {
+			root->remove_child(type_node);
+			continue;
+		}
```

Comment:
Unless I'm missing something, this will cause a leak, because the `type_node` is not freed nor referenced anywhere anymore. You should probably use `memdelete()` here.

Reply:
Indeed, I didn't consider this. Fixed.

Reply:
`remove_child()` is unnecessary when you do `memdelete()` btw

Reply:
That's TreeItem's remove child though, not Node's. Are you sure?

Reply:
Ah yes, checked the deconstructor and it does call it anyway, if there is a parent reference. I guess it makes no difference performance wise then.

Reply:
See #38074 ;)